pkg/cmd/controller/pipeline/pipelinerunner_controller_test.go

```diff
@@ -240,3 +253,29 @@ func getFreePort() (int, error) {
 	}()
 	return l.Addr().(*net.TCPAddr).Port, nil
 }
+
+func waitForOpenPort(host string, port int, timeOut time.Duration) error {
```

Comment:
Is there any reason to not use `httptest` instead of opening directly the ports?

https://golang.org/pkg/net/http/httptest/

Reply:
hmm, maybe just ignorance. How would that look like in this case? 

What's happening is that we select a "random" free port and pass it to the controller. The controller then starts its HTTP server waiting for requests. We then make a few HTTP requests to test various things.

The issue I am trying to fix is that the way the test was written, the test request would in some cases be made w/o the HTTP server being ready (at least that's what I believe atm). SO for this reason, I added this _waitForOpenPort_ method which will wait until the HTTP server is listening or the timeout occurs.